index.js

```diff
@@ -0,0 +1,365 @@
+//.
+//.   - If a short option name has an associated value, the value must be
+//.     provided as a separate argument (e.g. `['-x', '.js']`).
+//.
+//.   - Short option names must be provided as separate arguments. One can
+//.     support `tar -z -x -v -f filename` but not `tar -zxvf filename`.
+//.
+//. [Commander.js]: https://github.com/tj/commander.js
+//.
+//. ## Algorithm
+//.
+//. There are three inputs:
+//.
+//.   - a _specification_, a mapping from flag/option name to handler;
+//.   - an arbitrary _configuration value_; and
+//.   - an array of _arguments_.
+//.
+//. There are two outputs:
+//.
+//.   - a _configuration value_; and
+//.   - an array of _positional arguments_.
+//.
+//. The separator consists of two hyphens (`/^--$/`).
+//.
+//. Short flag/option names consist of a hyphen followed by a non-hyphen
+//. character (`/^-[^-]$/`).
+//.
+//. Long flag/option names consist of two hyphens followed by one or more
+//. characters (`/^--[^]+$/`).
+//.
+//. Arguments are processed from left to right. For each argument, there are
+//. three possibilities:
+//.
+//.   - The argument is the separator. Processing ends and the current
+//.     configuration value is returned along with the remaining arguments.
+//.
+//.   - The argument is a (short/long) flag/option name. It is looked up in the
+//.     specification. There are three possibilities:
+//.
+//.       - The specification does not contain such a key. Processing ends and
+//.         an error value is returned.
+//.
+//.       - The corresponding value is a `Flag (f)`. `f` is applied to the
+//.         current configuration value to produce a new configuration value,
+//.         and processing continues with the next argument.
+//.
+//.       - The corresponding value is an `Option (g)`. The next argument is
+//.         the option value. There are two possibilities:
+//.
+//.           - There are no more arguments. Processing ends and an error value
+//.             is returned.
+//.
+//.           - The validation function `g` is applied to the next argument.
+//.             There are two possibilities:
+//.
+//.               - `g` returns an error value. Processing ends and the error
+//.                 value is returned.
+//.
+//.               - `g` returns a function `f`. `f` is applied to the current
+//.                 configuration value to produce a new configuration value,
+//.                 and processing continues with the argument following the
+//.                 option value.
+//.
+//.   - The argument is neither the separator nor a (short/long) flag/option
+//.     name. Processing ends and the current configuration value is returned
+//.     along with this argument and the remaining arguments.
+//.
+//. If, after processing the final argument, processing has not already ended,
+//. the current configuration value is returned along with an empty array.
+
+'use strict';
+
+const {create, env} = require ('sanctuary');
```

Comment:
Oh. Interesting. The lib is on the top of the stack. :o

Reply:
I removed this dependency. We still depend on sanctuary-def, sanctuary-either, and sanctuary-pair.